Cannot build python3Packages.pecan

**Describe the bug**
Build failed during test phase.
last 10 lines of log:
```
       > pecan/tests/test_no_thread_locals.py::TestRestController::test_custom_method_type
       > pecan/tests/test_rest.py::TestRestController::test_basic_rest
       >   /nix/store/1xji37jwmnj0dkij6jlwwsccnqpw0d33-python3.8-webtest-2.0.32/lib/python3.8/site-packages/webtest/lint.py:359: WSGIWarning: Unknown REQUEST_METHOD: 'RESET'
       >     warnings.warn(
       >
       > -- Docs: https://docs.pytest.org/en/stable/warnings.html
       > =========================== short test summary info ============================
       > FAILED pecan/tests/test_jsonify.py::TestJsonifySQLAlchemyGenericEncoder::test_result_proxy
       > FAILED pecan/tests/test_jsonify.py::TestJsonifySQLAlchemyGenericEncoder::test_row_proxy
       > ============ 2 failed, 541 passed, 1 skipped, 27 warnings in 3.13s =============
```
full log (produced by nix log /nix/store/smrzwazd7hxy58rbmkrmlb4ml0bb4kiy-python3.8-pecan-1.4.0.drv) is available at: https://gist.github.com/applePrincess/3164282320c2067d86438c07e77564a8
**To Reproduce**
Steps to reproduce the behavior:
1. nix build nixpkgs#python3Packages.pecan

**Expected behavior**
A successful build

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

**Notify maintainers**
No maintainers set so CC to list of contributers instead.
@FRidh @prusnak @SuperSandro2000 @ryantm @dotlambda @Profpatsch @Mic92 @drewrisinger @adisbladis 

**Metadata**

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.10, NixOS, 21.11.20210612.d51b663 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `"nixos-20.09.4242.cee3ca37c68, nixos-unstable-21.11pre293089.1c2986bbb80"`
 - channels(applePrincess): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```
nixpkgs git commit: `ea12a8ea1742389348950c77e75097320bd9d2dd`
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/python-modules/pecan/default.nix
+++ b/pkgs/development/python-modules/pecan/default.nix
@@ -1,5 +1,6 @@
 { lib
 , fetchPypi
+, fetchpatch
 , buildPythonPackage
 , isPy27
 # Python deps
@@ -28,6 +29,14 @@ buildPythonPackage rec {
     sha256 = "4b2acd6802a04b59e306d0a6ccf37701d24376f4dc044bbbafba3afdf9d3389a";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "Support-SQLAlchemy-1.4x.patch";
+      url = "https://github.com/pecan/pecan/commit/a520bd544c0b02a02dbf692b8d6e2f7a503ee6d4.patch";
+      sha256 = "sha256-QCHRjwnpy8ndCvcuyE5Y65BybKYthJXDySUtmpJD8gY=";
+    })
+  ];
+
   propagatedBuildInputs = [
     logutils
     Mako
@@ -51,8 +60,9 @@ buildPythonPackage rec {
   ];
 
   meta = with lib; {
-    description = "Pecan";
-    homepage = "http://www.pecanpy.org/";
+    description = "WSGI object-dispatching web framework, designed to be lean and fast";
+    homepage = "https://www.pecanpy.org/";
     changelog = "https://pecan.readthedocs.io/en/latest/changes.html";
+    maintainers = with maintainers; [ applePrincess ];
   };
 }

